_docker/drupal/managed-platform/ansible/templates/dev/drupal/rhd.settings.php.j2

```diff
@@ -106,4 +105,104 @@ $config['rhd_disqus.disqussettings']['rhd_disqus_secret_key'] = "{{ drupal_disqu
 $config['rhd_disqus.disqussettings']['rhd_disqus_api_key'] = "{{ drupal_disqus_api_key }}";
 $config['rhd_disqus.disqussettings']['rhd_disqus_shortname'] = "{{ drupal_disqus_shortname }}";
 $config['rhd_disqus.disqussettings']['rhd_disqus_email_author_enabled'] = "{{ drupal_disqus_email_author_enabled }}";
-$config['rhd_disqus.disqussettings']['rhd_disqus_email_author_debug'] = "{{ drupal_disqus_email_author_debug }}";
\ No newline at end of file
+$config['rhd_disqus.disqussettings']['rhd_disqus_email_author_debug'] = "{{ drupal_disqus_email_author_debug }}";
+
+/*
+    MWES-2871: Adds support for using Memcached as the cache engine instead of the database.
+*/
+if($config['redhat_developers']['cache']['engine'] == 'memcached') {
+
+    $settings['cache']['default'] = 'cache.backend.memcache';
+    $settings['memcache']['servers'] = [];
+    {% for number in range(0, drupal_memcached_nodes) %}
+$settings['memcache']['servers'] += [
+       '{{ drupal_memcached_node_name_pattern | format(number) }}:11211' => 'default'
+    ];
+    {% endfor %}
+
+    $settings['memcache']['bins'] = ['default' => 'default'];
+    $settings['memcache']['key_prefix'] = '{{ drupal_deployment_dir }}_';
+    $settings['memcache']['options'] = [
+        Memcached::OPT_DISTRIBUTION => Memcached::DISTRIBUTION_CONSISTENT,
+        Memcached::OPT_COMPRESSION => FALSE,
+        Memcached::OPT_TCP_NODELAY => TRUE,
+        Memcached::OPT_LIBKETAMA_COMPATIBLE => TRUE,
+        Memcached::OPT_NO_BLOCK => FALSE
+    ];
+
+    /*
+        Load the memcached services
+    */
+    $settings['container_yamls'][] = DRUPAL_ROOT . '/sites/default/memcached.services.yml';
+
+    /*
+        Taken from: https://git.drupalcode.org/project/memcache/blob/8.x-2.x/README.txt
+    */
+
+    $class_loader->addPsr4('Drupal\\memcache\\', 'modules/contrib/memcache/src');
+    // Define custom bootstrap container definition to use Memcache for cache.container.
+    $settings['bootstrap_container_definition'] = [
```

Comment:
@fearlsgroove @jordanpagewhite Could you have a look at this part for me. I've copied this verbatim from the memcached module README.txt. I'm not totally up on the implications of this so just would like someone with better Drupal experience than I to +1 it.

cheers,

Rob

Reply:
Hey @robpblake . This looks good to me after reviewing what you have here and the module. I am not very familiar with configuring Memcache, particularly in recent history as I almost exclusively worked with Paas platforms at my last workplace, but I think it is fair to trust the docs of the Memcache module as the usage is so high, it has an active issue queue, and it is supported by 2 well-known Drupal organizations.

Reply:
@jordanpagewhite I'm more interested in what is happening to the Drupal container definition in `rhd.settings.php` when we have `memcached` enabled. If you're comfortable with it, then so am I. I just don't want to blindly apply things without really having a true grasp of what it is doing.

Lemme know if you're happy.

cheers,

Rob